Sources/performance-worker/WorkerService.swift

```diff
@@ -41,12 +51,41 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
       self.role = role
     }
 
-    init(server: GRPCServer) {
-      self.role = .server(server)
+    var server: GRPCServer? {
+      switch self.role {
+      case let .server(serverState):
+        return serverState.server
+      default:
+        return nil
+      }
     }
 
-    init(client: GRPCClient) {
-      self.role = .client(client)
+    mutating func serverStats(replaceWith newStats: ServerStats? = nil) -> ServerStats? {
+      switch self.role {
+      case var .server(serverState):
+        let stats = serverState.stats
+        if let newStats = newStats {
+          serverState.stats = newStats
+          self.role = .server(serverState)
+        }
+        return stats
+      default:
```

Comment:
It's best to switch explicitly over the different options, particularly with state machines, because if you add a new case later you might not correctly handle it (because the `default` case will catch it).